### PR TITLE
Repair creation of diffusive domain

### PR DESCRIPTION
**Pull request by shorvath-noaa**
Some updates to how t-route determines the diffusive domain and reads topobathy data

## Additions
**AbstractRouting.py**
- Update to searching upstream from user provided tailwater to create diffusive domain. This still stops at waterbodies
- Filters topobathy data upon reading to only include segments that are in diffusive domain

## Removals

-

## Changes
**diffusive_utils_v02.py**
- Ensure topobathy_bytw is still a dataframe after filtering by segment id

**LowerColorado_TX_v4/domain/coastal_domain_tw.yaml**
- Remove diffusive tailwaters that are not needed for LC runs.

## Testing

1.

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: src/troute-network/troute/AbstractRouting.py [hunk 1]
```diff
@@ -74,8 +72,10 @@ def read_parquet(file_path):
     
 
     '''
-    tbl = pq.read_table(file_path)
-    df = tbl.to_pandas().dropna(subset=['hy_id','relative_dist','Z','roughness'])
+    df = pd.read_parquet(file_path,
```
Comment (kumdonoaa) [inline src/troute-network/troute/AbstractRouting.py:75]: When I use the newest topodata with manning's value attached, I have this error:
image
It seems that the column names of the new topo data is different from the previous version: For example, here:
https://github.com/kumdonoaa/t-route/blob/ee4be4b549625392229cdff6a4e0fea7ce0026cb/src/troute-network/troute/AbstractRouting.py#L78

Comment (shorvath-noaa) [reply src/troute-network/troute/AbstractRouting.py:75]: Yeah, I preprocessed the raw topobathy parquet files to create a `roughness` column that is the average of `min_n` and `max_n`. Do you think we should just use `min_n`?

Comment (shorvath-noaa) [reply src/troute-network/troute/AbstractRouting.py:75]: Oh, this is actually an error about `relative_distance`. That was also preprocessed, changed from `relative_distance` to `relative_dist`.

Comment (shorvath-noaa) [reply src/troute-network/troute/AbstractRouting.py:75]: The problem is troute_test.parquet has different column names than the newest DEM parquet files. So we either need consistency across parquet files, or a more complicated method of reading files with different formats.

Comment (kumdonoaa) [reply src/troute-network/troute/AbstractRouting.py:75]: I think averaged manning's n may be ok for this functional test but I need to study more how min_n and max_n have been developed.  In the code I attached in the last line of my previous comment, I simply picked min_n to represent n.  I think we may have crosswalk table or something to filter different names and converge into common name.

**Review by kumdonoaa — APPROVED** [review]

**Pull request merged**